### PR TITLE
Custom network test support

### PR DESCRIPTION
**Pull request by CjHare**
Changes proposed in this pull request:
Adds the ability to run the snapshot-strategies test using a local networks file. 
This adds supports dev & CI testing of strategies prior to governance contract deployment on a public (public test) networkk.

### CODE REVIEW

**Comment by CjHare** [comment]
This setup I've got running with a Hardhat node (Ethereum chain) and deployment of the mainnet Multicall and a ERC20 token passing the default `erc20-balanceof` strategy with local changes (not included in the commit) to the exmaple.json.

Let me know what you think for the PR 🙂

File: src/utils/multicaller.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import { StaticJsonRpcProvider } from '@ethersproject/providers';
+import set from 'lodash.set';
+import { multicall } from '../utils';
+
+export default class Multicaller {
```
Comment (ChaituVR) [inline src/utils/multicaller.ts:5]: @CjHare Great work on PR! but multicall and Multicaller here causes duplicate code from snapshot.js, we should find some way to make them usable from one place only ( maybe try [patching snapshot.js](https://www.npmjs.com/package/patch-package) )

Comment (ChaituVR) [reply src/utils/multicaller.ts:5]: @bonustrack any ideas? or these can be here? 🤔

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/utils/multicaller.ts (hunk 1 above)
Comment (CjHare) [reply src/utils/multicaller.ts:5]: @ChaituVR you're quite right, the only way I could see avoiding the duplication would be to patch `snapshot.js` in some way. 
From a design perspective I can appreciate `snapshot.js` not needing such a change (as it's a change for the purpose of testing) ...but I can always start that conversation 🙂

**Comment by CjHare** [comment]
Closed off as I've messed up the Git history for the branch (and this PR was/is blocked on the `snapshot.js` changess

**Pull request closed without merging**